### PR TITLE
Improve mongo db importing and mocking

### PR DESCRIPTION
**Pull request by plars**
This improves the way we handle the db object, avoids needing to use current_app, and allows easier test db object creation

### CODE REVIEW

**Review by codersquid — APPROVED** [review]
lgtm though you might want to get an approval from someone who has more experience.

File: src/__init__.py [hunk 1]
```diff
@@ -174,31 +175,27 @@ def setup_mongodb(application):
...
         uuidRepresentation="standard",
         serverSelectionTimeoutMS=2000,
         socketTimeoutMS=10000,
     )
-    application.db = mongo_client.db
 
     # Initialize collections and indices in case they don't exist already
     # Automatically expire jobs after 7 days if nothing runs them
-    application.db.jobs.create_index(
+    mongo.db.jobs.create_index(
```
Comment (nadzyah) [inline src/__init__.py:188]: Is it okay that you don't remove the `db` field here and further?

File: tests/test_v1.py [hunk 2]
```diff
@@ -23,23 +23,38 @@
...
+
+
+class MongoClientMock(mongomock.MongoClient):
+    """Mock MongoClient and allow GridFS"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        enable_gridfs_integration()
+
+    def start_session(self, *args, **kwargs):
```
Comment (nadzyah) [inline tests/test_v1.py:37]: Isn't it better to disable pylint for specific lines instead of using dirty hacks?

**Review by nadzyah — CHANGES_REQUESTED** [review]
Thanks for your changes! Just minor things I guess we need to  improve :)

File: src/__init__.py (hunk 1 above)
Comment (plars) [reply src/__init__.py:188]: Yes, "mongo" here is a PyMongo object, which contains the db object.

File: tests/test_v1.py (hunk 2 above)
Comment (plars) [reply tests/test_v1.py:37]: I don't really have a strong opinion on whether it's dirtier to ignore the pylint warning or do this, however I think it's quite common that #noqa lines last forever and sometimes without a clear reason. Pylint was complaining that start_session() was an abstract method and that I wasn't implementing it. Looking at MongoClient, I don't *think* that seems to be true from what I'm seeing. However, I also see no harm in implementing the stub to make it happy. I don't call it directly, but perhaps something else might.

Comment (nadzyah) [reply tests/test_v1.py:37]: No problem, if you have a reason to keep it, let's try it!

**Review by nadzyah — APPROVED** [review]
+1

**Pull request merged**